app/code/community/Bolt/Boltpay/controllers/ApiController.php

```diff
@@ -162,6 +162,11 @@ public function hookAction()
             );
             $this->getResponse()->setHttpResponseCode(201);
 
+        } catch (Bolt_Boltpay_DuplicatedTransitionException $boltPayDuplicatedTransitionException) {
+            //In this case, The parent quote is currently being processed or has been processed,
+            //and this duplicated transaction should be ignored
+            Mage::helper('boltpay/bugsnag')->notifyException($boltPayDuplicatedTransitionException);
+            $this->getResponse()->setHttpResponseCode(200);
```

Comment:
I have a mixed feeling here. I'd say let's return error until we can be certain this only caused by harmless race condition

Reply:
Yes, that's what I want to discuss, in this situation, a previous webhook already started to create order, and it can determine whether success or failure then response properly, so this duplicated transition should be ignored, its response actually is useless, does bolt server support such a response code?

Reply:
well so if you can determine that hook is called with Bolt transaction reference that is already processed, you can return 200 here. 
However, if for some reason Bolt has two different transaction reference for the single order_reference (quote_id), we're in trouble and in that case plugin should return error so we can get notified and investigate.

Reply:
Done, I separate the ways to handle different situations